plugins/ci/pkg/ci/git.go

```diff
@@ -62,41 +78,31 @@ func getGitInfo(baseRepoPath, repoName, baseBranch string) (*gitInfo, error) {
 
 	branch := os.Getenv("BRANCH_NAME")
 	if branch == "" {
-		branch, err = commands.ExecInDir(baseRepoPath, exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD"), "getting branch name")
+		branch, err = cmdExecutor(baseRepoPath, exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD"), "getting branch name")
 		if err != nil {
-			logrus.Error("Unable to get GIT Branch Name")
-			return nil, err
+			logrus.Warnf("Unable to get GIT branch name: %v", err)
+			gitCommandFail = true
 		}
 	}
 	logrus.Infof("Branch: %s", branch)
 
 	origin := os.Getenv("ORIGIN_URL")
 	if origin == "" {
-		origin, err = commands.ExecInDir(baseRepoPath, exec.Command("git", "remote", "get-url", "origin"), "getting origin url")
+		origin, err = cmdExecutor(baseRepoPath, exec.Command("git", "remote", "get-url", "origin"), "getting origin url")
 		if err != nil {
-			logrus.Error("Unable to get GIT Origin")
-			return nil, err
+			logrus.Warnf("Unable to get GIT origin: %v", err)
+			gitCommandFail = true
 		}
 	}
 	logrus.Infof("Origin: %s", util.RemoveToken(origin))
 
+	if gitCommandFail {
+		logGitCIRunnerHint(ciRunner)
+	}
+
 	if repoName == "" {
-		logrus.Infof("No repositoryName set, defaulting to origin.")
-		repoName = origin
-		if strings.Contains(repoName, "@") { // git@github.com URLs are allowed
-			repoNameSplit := strings.Split(repoName, "@")
-			// Take the substring after the last @ to avoid any tokens in an HTTPS URL
-			repoName = repoNameSplit[len(repoNameSplit)-1]
-		} else if strings.Contains(repoName, "//") {
-			repoNameSplit := strings.Split(repoName, "//")
-			repoName = repoNameSplit[len(repoNameSplit)-1]
-		}
-		// Remove "******.com:" prefix and ".git" suffix to get clean $org/$repo structure
```

Comment:
These comments about what is being removed by splitting on a particular character are very helpful to understand why the code does what it does!

Reply:
Indeed - also adding tests, so we may clearly see all the expected inputs and their respected outputs